hoodie-client/src/main/java/com/uber/hoodie/io/HoodieIOHandle.java

```diff
@@ -104,4 +110,29 @@ public Schema getSchema() {
   public static Schema createHoodieWriteSchema(HoodieWriteConfig config) {
     return HoodieAvroUtils.addMetadataFields(new Schema.Parser().parse(config.getSchema()));
   }
+
+  // Dynamic calculation of maxMemory to use for merge
+  protected long getMaxMemoryAllowedForMerge() {
+
+    if (SparkEnv.get() != null) {
+      long executorMemoryInBytes = Long.valueOf(
+          Utils.memoryStringToMb(SparkEnv.get().conf().get(config.getSparkExecutorMemoryProp(),
+              config.getDefaultSparkExecutorMemory())) * 1024
+              * 1024L); // 1 GB is the default conf used by Spark, look at SparkContext.scala
+      double memoryFraction = Double
+          .valueOf(SparkEnv.get().conf().get(config.getSparkExecutorMemoryFractionProp(),
+              String.valueOf(config
+                  .getDefaultSparkExecutorMemoryFraction()))); // 0.6 is the default value used by Spark, look at SparkConf.scala
+      logger.info("MemoryFraction allowed for executor memory => " + memoryFraction);
+      double maxMemoryFractionForMerge = config.getMaxMemoryFractionPerPartitionMerge();
+      logger.info(
+          "MemoryFraction of executor memory allowed for merge => " + maxMemoryFractionForMerge);
+      long maxMemoryForMerge = (long) Math
+          .floor(executorMemoryInBytes * memoryFraction * maxMemoryFractionForMerge);
+      logger.info("MaxMemoryInBytes allowed for merge => " + maxMemoryForMerge);
```

Comment:
nit: can you group these log statements into a single one, that prints all different values? it will definitely be helpful while debugging to have this math in a single log line

Reply:
done